src/nix-build/nix-build.cc

```diff
@@ -447,7 +447,9 @@ static void main_nix_build(int argc, char * * argv)
                 "unset NIX_ENFORCE_PURITY; "
                 "shopt -u nullglob; "
                 "unset TZ; %6%"
-                "%7%",
+                "envCommand=(%7%);"
+                "[ ! -x \"${envCommand[1]}\" ] && exit 1;"
```

Comment:
```suggestion
                "[ ! -x \"${envCommand[1]}\" ] && exit 127;"
```

127 is commonly used in these situations.

Reply:
Fixed, thanks!